metadata_backend/services/datacite_service_handler.py

```diff
@@ -128,5 +129,36 @@ async def delete(self, doi: str) -> None:
         :param doi: identifier to be utilized for deleting draft DOI
         :raises: HTTPInternalServerError if we the Datacite draft DOI delete fails
         """
-        await self._request(method="DELETE", path=doi)
+        await self._request(method="DELETE", path=f"/dois/{doi}")
         LOG.info(f"Datacite doi {doi} deleted.")
+
+    async def heartbeat(self) -> Dict:
+        """Check DOI service hearthbeat.
+
+        This can return only 200 or 500
+
+        :returns: Dict with status of the datacite status
+        """
+        try:
+            start = time.time()
+            async with self._client.request(
+                method="GET",
+                url=f"{self.base_url}/heartbeat",
+                timeout=10,
+            ) as response:
+
+                content = await response.text()
+                LOG.info(f"Datacite REST API status is {content}.")
+                if response.status == 200 and content == "OK":
+                    status = "Ok" if (time.time() - start) < 1000 else "Degraded"
+                else:
+                    status = "Down"
+
+                return {"status": status}
+        except Exception as e:
+
+            LOG.info(f"Datacite REST API status retrieval failed with {e}.")
+            return {"status": "Error"}
+
+        finally:
+            await self._client.close()
```

Comment:
In the approach described above, we'd have one client session per service handler, and the client should be closed in the server teardown process, instead of after each request.